Bug 1522348 - Bulk assign open bugs to task, enhancement, defect field

kyoshino opened the pull request:
Happy SQL time! Add a script to do the migration task 1, 2, 3 written in [Bug 1524738](https://bugzilla.mozilla.org/show_bug.cgi?id=1524738#c6). Would like to test this on the dev server.

## Bugzilla link

[Bug 1522348 - Bulk assign open bugs to task, enhancement, defect field](https://bugzilla.mozilla.org/show_bug.cgi?id=1522348)

dylanwh commented:
I'll run this against my database dump as soon as I get it

dylanwh commented on scripts/migrate-bug-type.pl:
```diff
@@ -0,0 +1,148 @@
+use Bugzilla;
+
+# List of products and components that use a bug type other than "defect"
+use constant MIGRATION_MAP => (
```
```suggestion
my @MIGRATION_MAP => (
```

kyoshino replied:
Thanks 😸

kyoshino commented:
I'm moving this script to `/extensions/BMO/bin/` since this is BMO-specific.

kyoshino commented:
Hmm this time the sanity test warns an untouched script 🤔 
```
#   Failed test 'extensions/BMO/bin/migrate-github-pull-requests.pl DOES NOT require Perl 5.10.1 --WARNING'
#   at t/002goodperl.t line 109.

#   Failed test 'extensions/BMO/bin/migrate-bug-type.pl DOES NOT require Perl 5.10.1 --WARNING'
#   at t/002goodperl.t line 109.
```

kyoshino commented:
@dylanwh do you have any idea why these tests are failing. I’ve just moved the file under `extensions`.

dklawren commented:
> 
> 
> @dylanwh do you have any idea why these tests are failing. I’ve just moved the file under `extensions`.

Since the file is not explicitly using Mojolicioius (which automatically use 5.10.1), you need to add:

use 5.10.1;

to the top of the file.

kyoshino commented:
@dklawren Thanks! Somehow I was removing `use 5.10.1` from `migrate-github-pull-requests.pl` which made me confused myself. Added `use 5.10.1` to `migrate-bug-type.pl` so this should be fine now.

Speaking of the unexpected results, I’ve added a `if (scalar @$bug_ids)` check in case no bugs found in any listed component. Tested locally with a limited set of bugs/products/components and it worked. @dylanwh: can you please re-run this on the staging server?

kyoshino commented:
Test done on the staging server.

kyoshino commented:
We gotta run it later this week.

kyoshino commented:
I’m checking if any of the recently added or removed components should be listed here to use the enhancement or task type. Cc @emceeaich

kyoshino commented:
Good to go.

kyoshino commented on scripts/migrate-bug-type.pl:
```diff
@@ -0,0 +1,148 @@
+
+$dbh->bz_start_transaction;
+
+# Disable the "enhancement" severity
```
Thanks, but somehow this says "outdated suggestions cannot be applied" so I'll make the change manually now.
